packages/omgidl-parser/src/IDLNodeProcessor.ts

```diff
@@ -131,20 +132,44 @@ export class IDLNodeProcessor {
 
       // need to iterate through keys because this can occur on arrayLength, upperBound, arrayUpperBound, value, defaultValue
       for (const key of POSSIBLE_UNRESOLVED_MEMBERS) {
-        const value = node[key];
-        if (typeof value !== "object") {
+        const keyValue = node[key];
+        let finalKeyValue = undefined;
+        if (Array.isArray(keyValue)) {
+          const arrayLengths = keyValue;
+
+          let arrayLengthsResolved = false;
+          const finalArrayLengths = arrayLengths.map((arrayLength) => {
+            if (typeof arrayLength === "number") {
+              return arrayLength;
+            }
+            arrayLengthsResolved = true;
+            const constantNode = this.resolveConstantReference({
+              constantName: arrayLength.name,
+              nodeScopedIdentifier: scopedIdentifier,
+            });
+            return constantNode.value;
+          });
+          if (arrayLengthsResolved) {
```

Comment:
I don't quite get the purpose of this boolean – why do we need to _conditionally_ replace finalKeyValue instead of always doing it?

Also do we need to somehow check that resolveConstantReference was actually able to resolve the reference? Or will it throw if it fails?

Reply:
> conditionally replace `finalKeyValue`

Yeah this wasn't necessary but it follows the same behavior of the other constant values of not setting the finalKeyValue when constants aren't being used. Saves us from setting the value again. I've removed it since it seems like it was adding some confusion.

> check that resolveConstantReference resolves

Yeah it throws if it can't find the constant. So that's covered.